metrics/prometheus_exporter.go

```diff
@@ -43,7 +43,11 @@ func (emptyPromExporter) ExportView(viewData *view.Data) {
 
 //nolint: unparam // False positive of flagging the second result of this function unused.
 func newPrometheusExporter(config *metricsConfig, logger *zap.SugaredLogger) (view.Exporter, ResourceExporterFactory, error) {
-	e, err := prom.NewExporter(prom.Options{Namespace: config.component})
+	opts := prom.Options{}
+	if config.enableDeprecatedMetricPrefix {
+		opts.Namespace = config.component
```

Comment:
Why we set the Namespace to be equal to a component name here?

Reply:
this is the legacy behaviour - which I want to preserve with this flag